plugins/inputs/azure_monitor/azure_monitor.go

```diff
@@ -158,8 +160,17 @@ func (acm *azureClientsManager) createAzureClients(
 	clientID string,
 	clientSecret string,
 	tenantID string,
-) (*receiver.AzureClients, error) {
-	azureClients, err := receiver.CreateAzureClients(subscriptionID, clientID, clientSecret, tenantID)
+) (azureClients *receiver.AzureClients, err error) {
+	var token azcore.TokenCredential
+	if clientSecret != "" {
+		azureClients, err = receiver.CreateAzureClients(subscriptionID, clientID, clientSecret, tenantID)
+	} else {
+		if token, err = azidentity.NewDefaultAzureCredential(&azidentity.DefaultAzureCredentialOptions{
+			TenantID: tenantID,
+		}); err == nil {
+			azureClients, err = receiver.CreateAzureClientsWithCreds(subscriptionID, token)
+		}
+	}
 	if err != nil {
 		return nil, fmt.Errorf("error creating Azure clients: %w", err)
 	}
```

Comment:
Can we please avoid named return variables wherever possible? This causes scope confusion in more complex code...

```suggestion
) (*receiver.AzureClients, error) {
	if clientSecret != "" {
		return receiver.CreateAzureClients(subscriptionID, clientID, clientSecret, tenantID)
	}
	
	token, err := azidentity.NewDefaultAzureCredential(&azidentity.DefaultAzureCredentialOptions{TenantID: tenantID})
	if err != nil {
		return nil, fmt.Errorf("error creating Azure token: %w", err)
	}
	return receiver.CreateAzureClientsWithCreds(subscriptionID, token)
```